nannou/src/app.rs

```diff
@@ -1175,6 +1175,12 @@ fn run_loop<M, E>(
                             window_rect,
                         );
 
+                        // Clear the raw frame immediately once the window is invalidated
+                        if window.is_invalidated && frame_data.is_some() {
+                            let data = frame_data.as_ref().expect("missing `frame_data`");
+                            raw_frame.clear(&data.render.texture_view(), window.clear_color);
+                        }
```

Comment:
Maybe you can use an `if let` to get rid of the `.expect()`?

```suggestion
                        if window.is_invalidated {
                            if let Some(data) = frame_data {
                                raw_frame.clear(&data.render.texture_view(), window.clear_color);
                            }
                        }
```

Reply:
Thanks for taking a look! The downside is that it adds nesting -- which I typically try to avoid. But, I too suspect that it might be more canonical. So, I just tested & re-committed it with the change.